src/main/java/com/example/ajaxonboardingproject/controller/ShoppingCartController.kt

```diff
@@ -31,16 +31,16 @@ data class ShoppingCartController(
     ) {
         val details = auth.principal as UserDetails
         val email: String = details.username
-        val user: User = userService.findByEmail(email)
-        val movieSession: MovieSession = movieSessionService.get(movieSessionId)
+        val user: User = userService.findByEmail(email).block()!!
+        val movieSession: MovieSession = movieSessionService.get(movieSessionId).block()!!
         shoppingCartService.addSession(movieSession, user.id)
     }
 
     @GetMapping("/by-user")
     fun getByUser(auth: Authentication): ShoppingCartResponseDto {
         val details = auth.principal as UserDetails
         val email: String = details.username
-        val user: User = userService.findByEmail(email)
-        return shoppingCartResponseDtoMapper.mapToDto(shoppingCartService.getShoppingCartByUser(user.id))
+        val user: User = userService.findByEmail(email).block()!!
+        return shoppingCartResponseDtoMapper.mapToDto(shoppingCartService.getShoppingCartByUser(user.id).block()!!)
```

Comment:
```suggestion
        return userService.findByEmail(email) 
                        .flatMap { user -> shoppingCartService.getShoppingCartByUser(user.id) }
                        .map(shoppingCartResponseDtoMapper::mapToDto)
                        .block()!!
```

sorry for formatting

Reply:
please, fix all places

Reply:
Done!